### ISSUE TITLE
Missing Event/type ID, event.id is very ambiguous

### ISSUE THREAD

**Issue by vbohata** [elastic/ecs#306]
ECS field event.id usage is very unclear. Its description "Unique ID to describe the event" can mean it should uniquely describe event type/kind or the event itself should be unique to all other events. Usually (in Windows event log, SIEM solutions, ...) it identifies event kind/type/id. In this case there is a missing ECS field to uniquely identify the event in all other events. In other case there is a missing ECS field for identification of event id as used for example in windows event log - so event XY means it is security event from ...

**Comment by webmat** [comment]
I agree this needs to be addressed.

Note that in migrating Winlogbeat to ECS (https://github.com/elastic/beats/pull/10333), I mapped the Windows event ID to `event.code`. It's not officially in ECS yet, but I'm thinking this would make sense.

**Comment by vbohata** [comment]
OK. What about record numbers and event.id? Maybe there should also be integer log.record_number field, very useful for sorting logs?
So:
- event.code for "types" of events like Windows Event ID
- log.record_number for line number or windows event record number
- event.id keep for cases where previous two are unusable

**Comment by willemdh** [comment]
Just jumping in to share some event / type / message id use cases. In a lot of logs, there is a sort of event. code / event.id / message.id field.

For example in our Brocade audit logs:

```
0 AUDIT, 2005/12/10-09:54:03, [SEC-1000], WARNING, SECURITY,
JohnSmith/root/192.0.2.2/Telnet/CLI, Domain A/JohnsSwitch, , Incorrect password
during login attempt.
```

![image](https://user-images.githubusercontent.com/6462991/56020448-7d4f1600-5d07-11e9-95a1-b231d3be6e42.png)

So in this case `SEC-1000` is the event id, but as event.id in ECS needs to be unique, something like event.code would be very useful. Same goes for other type of logs (such as F5)

**Comment by ebeahan** [comment]
`event.code` was added in #439 and released in [1.1](https://github.com/elastic/ecs/releases/tag/v1.1.0).

I'm going to close this out. If there are any follow-up discussion points, please feel free to open a new issue 😄
